src/components/calendarInput/index.jsx

```diff
@@ -0,0 +1,112 @@
+import CalendarIcon from 'assets/calendar.png';
+import { addDays, addYears, endOfYear, format, startOfMonth } from 'date-fns';
+import React, { useEffect, useRef, useState } from 'react';
+import { DayPicker } from 'react-day-picker';
+import { useDispatch, useSelector } from 'react-redux';
+import { setDate } from 'store/slices/inputsSlice';
+
+import { MainInput, MainInputImg, MainInputWrapper } from '../common/styled';
+import { DayPickerWrapper } from './styled';
+
+const DATE_FORMAT_PATTERN = 'iii d MMM';
+const today = new Date();
+
+export default function CalendarInput() {
+  const dates = useSelector(state => state.inputs.dates);
+  const [range, setRange] = useState(() => ({
+    from: dates.from ? new Date(dates.from) : null,
+    to: dates.to ? new Date(dates.to) : null,
+  }));
+  const [showCalendar, setShowCalendar] = useState(false);
+  const dispatch = useDispatch();
+
+  const notInitialRender = useRef(false);
+
+  useEffect(() => {
+    if (notInitialRender.current)
+      dispatch(
+        setDate({
+          from: range.from ? format(range.from, 'y-M-d') : null,
+          to: range.to ? format(range.to, 'y-M-d') : null,
+        }),
+      );
+    else notInitialRender.current = true;
```

Comment:
скобки спасают нервы и жизни, крайне рекомендую
```suggestion
    if (notInitialRender.current) {
      dispatch(
        setDate({
          from: range.from ? format(range.from, 'y-M-d') : null,
          to: range.to ? format(range.to, 'y-M-d') : null,
        }),
      );
    } else {
      notInitialRender.current = true;
    }
```

Reply:
Ok. Добавил